plugins/cody-chat/src/com/sourcegraph/cody/workspace/WorkbenchListener.java

```diff
@@ -89,9 +91,16 @@ public void partOpened(IWorkbenchPartReference partReference) {
 
   @Override
   public void dispose() {
-    for (var child : children) {
-      child.dispose();
-    }
-    PlatformUI.getWorkbench().getActiveWorkbenchWindow().getPartService().removePartListener(this);
+    Display.getDefault()
+        .execute(
```

Comment:
This works slightly differently than `asyncExec`, which adds runnable to the queue to be run on the UI thread once it is free.

`execute`, on the other hand, runs the task immediately if it is already on the UI thread. If called from another thread, it falls back to `asyncExec`.

It is used when the code is already expected to be called from the UI thread.